woocommerce/woocommerce
# Show feedback footer on product editor page

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #38519.

#### No footer (ignore section backgrounds; see #38607)

<img width="1115" alt="Screenshot 2023-06-05 at 19 35 07" src="https://github.com/woocommerce/woocommerce/assets/2098816/d180e3ef-a944-4b07-9f29-ec092f1adc80">

#### Footer

<img width="1115" alt="Screenshot 2023-06-05 at 19 37 06" src="https://github.com/woocommerce/woocommerce/assets/2098816/700c43c7-57cb-45e0-bcc0-a65c07325ebd">

#### Footer with overlapping transient notice (to be addressed outside of this PR)

<img width="1115" alt="Screenshot 2023-06-05 at 19 36 54" src="https://github.com/woocommerce/woocommerce/assets/2098816/6421a525-fd8a-4299-936d-2f9a1a26d063">



### How to test the changes in this Pull Request:



Note: To reset the state of the system so that the feedback footer will be shown, delete the following options:
* `woocommerce_product_editor_show_feedback_bar`
* `woocommerce_ces_shown_for_actions`

1. Enable experimental "new product editor" (beta) in WooCommerce > Settings > Advanced > Features
2. Go to Products > Add New
3. Verify no feedback footer is shown.
4. Fill out some fields and save draft (or publish).
5. Verify feedback footer is shown.
6. Refresh the page.
7. Verify feedback footer is still shown.

Note: For the following, you will want to reset the state of the system between verifications to get the feedback footer to show (see above).

8. Verify that if the "X" button is clicked:
    - the feedback footer is hidden permanently
    - the `wcadmin_product_editor_feedback_bar_dismiss_click` Tracks event is logged
9. Verify that if the "Share feedback" link is clicked:
    - the standard CES feedback modal is shown (exact style will be handled in #38502 and #38521)
    - the feedback footer is hidden permanently
    - the `wcadmin_product_editor_feedback_bar_share_feedback_click` Tracks event is logged
10. Verify that if the "turn if off" link is clicked:
    - a custom feedback modal is shown (exact style will be handled in #38592 and #38521)
    - the feedback footer is hidden permanently
    - the `wcadmin_product_editor_feedback_bar_turnoff_editor_click` Tracks event is logged
